Libraries/LibGfx/Bitmap.cpp

```diff
@@ -61,6 +61,13 @@ RefPtr<Bitmap> Bitmap::create_purgeable(BitmapFormat format, const Size& size)
     return adopt(*new Bitmap(format, size, Purgeable::Yes));
 }
 
+RefPtr<Bitmap> Bitmap::cloned_from(RefPtr<Bitmap> bitmap)
+{
+    auto cloned_bitmap = Bitmap::create(bitmap->format(), bitmap->size());
+    memcpy(cloned_bitmap->scanline(0), bitmap->scanline(0), bitmap->size_in_bytes());
+    return cloned_bitmap;
```

Comment:
I wonder if there's a way to use virtual memory magic to avoid the actual copy.

Reply:
This was the suggestion that @awesomekling gave to me over IRC as he was teaching me some things on how these things work. Maybe there's other ways of having that memory writable. Indeed it would be more efficient

Reply:
I don't think there's any standard Unix/POSIX APIs that let you do what I want. On Linux, I bet you can `mmap("/proc/self/mem")` to achieve it. Maybe we should add a custom syscall, say `copy_mmap(void* old, size_t size)`?

Reply:
I don't quite understand this kind of things but shouldn't we just don't seal() SharedBuffers we could want to write later. From what I talked with Andreas it seemed to be the problem.

Like making a call specifically to allow writing on SharedBuffer's contents isn't it __hammering it__? Like, it's sealed for some reason.

Again, don't know what I'm saying, just trying to tie things together in my head. Still learning!.

PS: Andreas is tagged in here so he probably will answer you better :sweat_smile:

Reply:
Don't worry, I don't understand what you're saying either 😀

All I'm saying is it would be nice to awoid the physical copy (while still making a virtual copy) using CoW semantics, with some help from the kernel.

Reply:
Let's leave that as a nice-to-have future improvement for now :)